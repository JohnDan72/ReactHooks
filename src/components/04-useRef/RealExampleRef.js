import React, { useState } from 'react';
import { Button, Divider } from 'rsuite';
// import PropTypes from 'prop-types';
import MultipleCustomHooks from "../03-examples/MultipleCustomHooks";
const propTypes = {};
const defaultProps = {};


const RealExampleRef = () => {
    
    const [show, setShow] = useState(false);


    return (
        <>
            <h3>Use Ref</h3>

            <Divider></Divider>
            {/* <Icon as={RiReactjsLine} spin size="4em" /> */}

            { show && <MultipleCustomHooks /> }

            <Button className="my-5"
                    appearance="default"
                    onClick={()=>setShow(!show)}>Show/Hide</Button>
            
        </>
    );
}

RealExampleRef.propTypes = propTypes;
RealExampleRef.defaultProps = defaultProps;

export default RealExampleRef;