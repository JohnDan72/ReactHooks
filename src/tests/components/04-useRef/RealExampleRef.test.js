import { shallow } from "enzyme"
import RealExampleRef from "../../../components/04-useRef/RealExampleRef";


describe('Pruebas para RealExampleRef', () => {

    let wrapper = shallow(<RealExampleRef />);

    beforeEach(() => {
        wrapper = shallow(<RealExampleRef />);
    });


    test('debe renderizar el componente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);

    });


    test('debe mostrar el componente <MultipleCustomHooks/>', () => {
        const btn = wrapper.find('Button');
        btn.simulate('click');
        const multiple = wrapper.find('MultipleCustomHooks');

        expect(wrapper).toMatchSnapshot();
        expect(multiple.exists()).toBe(true);

    })


})