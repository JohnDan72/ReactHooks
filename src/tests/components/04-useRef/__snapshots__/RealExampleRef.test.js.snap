// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para RealExampleRef debe mostrar el componente <MultipleCustomHooks/> 1`] = `
<Fragment>
  <h3>
    Use Ref
  </h3>
  <Divider />
  <MultipleCustomHooks />
  <Button
    appearance="default"
    className="my-5"
    onClick={[Function]}
  >
    Show/Hide
  </Button>
</Fragment>
`;

exports[`Pruebas para RealExampleRef debe renderizar el componente 1`] = `
<Fragment>
  <h3>
    Use Ref
  </h3>
  <Divider />
  <Button
    appearance="default"
    className="my-5"
    onClick={[Function]}
  >
    Show/Hide
  </Button>
</Fragment>
`;
